services/web/server/src/simcore_service_webserver/login/_registration.py

```diff
@@ -0,0 +1,239 @@
+        data=data_model.json(),
+    )
+    return confirmation
+
+
+async def check_and_consume_invitation(
+    invitation_code: str, db: AsyncpgStorage, cfg: LoginOptions
+) -> InvitationData:
+    """Consumes invitation: the code is validated, the invitation retrieives and then deleted
+       since it only has one use
+
+    If valid, it returns InvitationData, otherwise it raises web.HTTPForbidden
+
+    :raises web.HTTPForbidden
+    """
+    if confirmation_token := await validate_confirmation_code(invitation_code, db, cfg):
+        try:
+            invitation = _InvitationValidator.parse_obj(confirmation_token)
+            return invitation.data
+
+        except ValidationError as err:
+            log.warning(
+                "%s is associated with an invalid %s.\nDetails: %s",
+                f"{invitation_code=}",
+                f"{confirmation_token=}",
+                f"{err=}",
+            )
+
+        finally:
+            await db.delete_confirmation(confirmation_token)
+            log.info("Invitation with %s was consumed", f"{confirmation_token=}")
+
+    raise web.HTTPForbidden(
+        reason=(
+            "Invalid invitation code."
+            "Your invitation was already used or might have expired."
+            "Please contact our support team to get a new one."
+        ),
+        content_type=MIMETYPE_APPLICATION_JSON,
+    )
+
+
+def get_confirmation_info(
+    cfg: LoginOptions, confirmation: ConfirmationTokenDict
+) -> ConfirmationTokenInfoDict:
+    """
+    Extends ConfirmationTokenDict by adding extra info and
+    deserializing action's data entry
+    """
+    info = ConfirmationTokenInfoDict(**confirmation)
+
+    action = ConfirmationAction(confirmation["action"])
+    if (data_type := ACTION_TO_DATA_TYPE[action]) and (data := confirmation["data"]):
+        info["data"] = parse_raw_as(data_type, data)
+
+    # extra
+    info["expires"] = get_expiration_date(cfg, confirmation)
+
+    if confirmation["action"] == ConfirmationAction.INVITATION.name:
+        info["url"] = f"{get_invitation_url(confirmation)}"
+
+    return info
+
+
+def get_invitation_url(
+    confirmation: ConfirmationTokenDict, origin: Optional[URL] = None
+) -> URL:
+    """Creates a URL to invite a user for registration
+
+    This URL is sent to the user via email
+
+    The user clicks URL link and ends up in the front-end
+
+    This URL appends a fragment for front-end that interprets as open registration page
+    and append the invitation code in the API request body together with the data added by the user
+    """
+    code = confirmation["code"]
+    is_invitation = confirmation["action"] == ConfirmationAction.INVITATION.name
+
+    if origin is None or not is_invitation:
+        origin = URL()
+
+    # https://some-web-url.io/#/registration/?invitation={code}
+    # NOTE: Uniform encoding in front-end fragments https://github.com/ITISFoundation/osparc-simcore/issues/1975
```

Comment:
What does this note mean, is this a todo? "NOTE" is very generic, potentially this can be subsituted with something that makes clear what this link is without clicking it (`feature request`, `future improvement`, `for implementation details see:`, or something like this)

Reply:
This is a convention we have in osparc-simcore. From some months ago, we decided that all ``TODO:`` and ``FIXME:`` cannot be merged into master. One of our bots detect those and flag them as code-smells. 
If we still want to keep this idea/fix noted in the code, those have to be rewritten as ``NOTE:`` AND must include a link to a github issue where we include the info you mention.

Reply:
Your question is justified, so I started adding these "non-written" conventions in ``docs/coding-conventions.md`` so we keep it for future developers.

Reply:
I am happy it makes you laugh ... but you got it half way right. This is a recurrent topic (which is mostly the part you are missing) and this was a compromise that made all the parts equally unhappy.  It is a long story but  you are really interested, I can tell you more details offline

Reply:
ok all good at least you know what you are doing or how this can be interpreted :D 

thanks for the PR! All good to me then

Reply:
@mrnicegyu11 if you have no other objections, you need to approve the PR. Otherwise it will remain blocked (since you rejected it in the first time)